apollo-federation/src/query_plan/fetch_dependency_graph.rs

```diff
@@ -2094,29 +2094,34 @@ impl FetchDependencyGraph {
     fn can_merge_grand_child_in(
         &self,
         node_id: NodeIndex,
-        grand_child_id: NodeIndex,
+        child_id: NodeIndex,
```

Comment:
@duckki It should AFAIK, but the JS code doesn't technically assert this. I was writing it this way to keep in line with the JS code, with the intention of adding an assert later. (At the moment I'm trying to find discrepancies between JS and Rust rather than simplify code, IMO that can come later.)

Reply:
(It's also worth noting, it's not a guarantee that the grand child passed into `can_merge_grand_child_in()` is actually a grand child, that's why I've changed the argument name to `maybe_grand_child_id`. It's only really checked after the first 10 lines.)

Reply:
nice catch, it looks like it was missed when the checks for `inputs` and `mergeAt` were moved here (as we were already getting some node weights).